docs/rules/no-deprecated-delete-set.md

```diff
@@ -0,0 +1,49 @@
+---
+pageClass: rule-details
+sidebarDepth: 0
+title: vue/no-deprecated-delete-set
+description: disallow using deprecated `$delete` and `$set` (in Vue.js 3.0.0+)
+---
+
+# vue/no-deprecated-delete-set
+
+> disallow using deprecated `$delete` and `$set` (in Vue.js 3.0.0+)
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> _**This rule has not been released yet.**_ </badge>
+- :gear: This rule is included in all of `"plugin:vue/vue3-essential"`, `*.configs["flat/essential"]`, `"plugin:vue/vue3-strongly-recommended"`, `*.configs["flat/strongly-recommended"]`, `"plugin:vue/vue3-recommended"` and `*.configs["flat/recommended"]`.
+
+## :book: Rule Details
+
+This rule reports use of deprecated `$delete` and `$set`. (in Vue.js 3.0.0+).
+
+<eslint-code-block :rules="{'vue/no-deprecated-delete-set': ['error']}">
+
+```vue
+<script>
+  import { set as st, delete as del } from 'vue'
```

Comment:
There was no `delete` named export in Vue 2.7 (since that is a registered keyword in JavaScript), it was called `del` directly. Unfortunately, I cannot find a source for that, only the Composition API plugin for Vue 2.6 mentions it: https://github.com/vuejs/composition-api?tab=readme-ov-file#reactive

And as an example in the docs, I think it doesn't make sense to rename `set` to `st`. Of course it should be supported and tested, just not shown upfront in the docs:

```suggestion
  import { set, del } from 'vue'
```

Reply:
I see! Looked into it but couldn't find the export `del`, now makes sense to me.